src/components/settings/data/hooks.js

```diff
@@ -60,7 +61,59 @@ export const useLinkManagement = (enterpriseUUID) => {
   };
 };
 
+const initialCustomerAgreementState = {
+  netDaysUntilExpiration: 0,
+};
+/*
+ * @param {Objet {enterpriseId: string}}
+ * @returns {customerAgreement: Object, loadingCustomerAgreement: bool}
+ * customerAgreement:{
+ *  netDaysUntilExpiration: number
+ * }
+ */
+export const useCustomerAgreementData = ({ enterpriseId }) => {
+  const [customerAgreement, setCustomerAgreement] = useState(initialCustomerAgreementState);
+  const [loadingCustomerAgreement, setLoadingCustomerAgreement] = useState(true);
+
+  const loadCustomerAgreementData = (page = 1) => {
+    if (!loadingCustomerAgreement) {
+      setLoadingCustomerAgreement(true);
+    }
+    LicenseManagerApiService.fetchCustomerAgreementData({ enterprise_customer_uuid: enterpriseId, page })
```

Comment:
Preferably use `async`/`await`, but it's kinda personal preference, so will leave it up to you if you want to address :)

Reply:
Yes! 🚀